yt_dlp/networking/_curlcffi.py

```diff
@@ -149,6 +149,7 @@ def _check_extensions(self, extensions):
         # CurlCFFIRH ignores legacy ssl options currently.
         # Impersonation generally uses a looser SSL configuration than urllib/requests.
         extensions.pop('legacy_ssl', None)
+        extensions.pop('keep_header_casing', None)
```

Comment:
```suggestion
```

curl cffi does not support keeping header casing sometimes (e..g http/2 from memory)